src/Stratis.Bitcoin.Features.Consensus/PowConsensusValidator.cs

```diff
@@ -818,18 +973,29 @@ private void MerkleComputation(List<uint256> leaves, ref uint256 root, ref bool
             root = hh;
         }
 
+        /// <summary>
+        /// Gets index of the last coinbase transaction output with SegWit flag.
+        /// </summary>
+        /// <param name="block">Block which coinbase transaction's outputs will be checked for SegWit flags.</param>
+        /// <returns>
+        /// <c>-1</c> if no SegWit flags were found.
+        /// If SegWit flag is found index of the last transaction's output that has SegWit flag is returned.
+        /// </returns>
         private int GetWitnessCommitmentIndex(Block block)
         {
             int commitpos = -1;
             for (int i = 0; i < block.Transactions[0].Outputs.Count; i++)
             {
-                if ((block.Transactions[0].Outputs[i].ScriptPubKey.Length >= 38) &&
-                    (block.Transactions[0].Outputs[i].ScriptPubKey.ToBytes(true)[0] == (byte)OpcodeType.OP_RETURN) &&
-                    (block.Transactions[0].Outputs[i].ScriptPubKey.ToBytes(true)[1] == 0x24) &&
-                    (block.Transactions[0].Outputs[i].ScriptPubKey.ToBytes(true)[2] == 0xaa) &&
-                    (block.Transactions[0].Outputs[i].ScriptPubKey.ToBytes(true)[3] == 0x21) &&
-                    (block.Transactions[0].Outputs[i].ScriptPubKey.ToBytes(true)[4] == 0xa9) &&
-                    (block.Transactions[0].Outputs[i].ScriptPubKey.ToBytes(true)[5] == 0xed))
+                var scriptPubKey = block.Transactions[0].Outputs[i].ScriptPubKey;
+                byte[] scriptBytes = scriptPubKey.ToBytes(true);
+
+                if ((scriptPubKey.Length >= 38) &&
+                    (scriptBytes[0] == (byte)OpcodeType.OP_RETURN) &&
+                    (scriptBytes[1] == 0x24) &&
+                    (scriptBytes[2] == 0xaa) &&
+                    (scriptBytes[3] == 0x21) &&
+                    (scriptBytes[4] == 0xa9) &&
+                    (scriptBytes[5] == 0xed))
```

Comment:
the original code is not equivalent with the new one
however, i'm not familiar with this code enough to be able to decide whether it matters or not
the difference is that if length of the sPK is less than 37, you don't call `.ToBytes`, which saves time 

on the other hand your code saves time by only doing `ToBytes` once

so it may be worse or better than original depending on how often the first condition is satisfied
best option here is to probably separate that first condition and only call ToBytes once only then
in which case we will have always better code